contracts/src/v0.8/ccip/pools/SiloedLockReleaseTokenPool.sol

```diff
@@ -0,0 +1,286 @@
+  /// The rebalancer will need to be set separately.
+  function updateSiloDesignationForChainSelectors(
+    uint64[] calldata removes,
+    ChainSiloConfigUpdate[] calldata adds
+  ) external onlyOwner {
+    for (uint256 i = 0; i < removes.length; ++i) {
+      // When a chain is removed from siloing, the funds are moved to the accounting pool shared by all unsiloed chain.
+      uint256 amountUnsiloed = s_siloedChainConfigs[removes[i]].siloedBalance;
+
+      s_unsiloedTokenBalance += amountUnsiloed;
+
+      /// @notice Removing the entire configuration will also delete the rebalancer. If the chain ever becomes siloed
+      /// again, the rebalancer will need to be set again.
+      delete s_siloedChainConfigs[removes[i]];
+
+      // Emit a removal event which includes the amount of funds moved to the general silo.
+      emit ChainUnsiloed(removes[i], amountUnsiloed);
+    }
+
+    for (uint256 i = 0; i < adds.length; ++i) {
+      /// @notice Adding a chain to siloing will not set the rebalancer
+      s_siloedChainConfigs[adds[i].remoteChainSelector].isSiloed = true;
+      s_siloedChainConfigs[adds[i].remoteChainSelector].rebalancer = adds[i].rebalancer;
+
+      // Emit a siloing event to indicate that the chain is now siloed, but that the amount of liquidity
+      // is zero until liquidity is provided.
+      emit ChainSiloed(adds[i].remoteChainSelector);
+    }
+  }
+
+  /// @notice Gets the rebalancer able to provide liquidity for a remote chain selector
+  /// @param remoteChainSelector The CCIP specific selector for the remote chain being interacted with.
+  /// @return The current liquidity manager, contract owner if the chain's funds are not siloed.
+  function getRebalancerByChain(
+    uint64 remoteChainSelector
+  ) public view returns (address) {
+    ChainSiloConfig memory remoteConfig = s_siloedChainConfigs[remoteChainSelector];
+    if (remoteConfig.isSiloed) return remoteConfig.rebalancer;
+    else return s_unsiloedChainRebalancer;
+  }
+
+  /// @notice Sets the Rebalancer address for a given remoteChainSelector.
+  /// @dev Only callable by the owner.
+  /// @param remoteChainSelector the remote chain to set.
+  /// @param newRebalancer the address allowed to add liquidity for the given siloed chain.
+  function setSiloedChainRebalancer(uint64 remoteChainSelector, address newRebalancer) external onlyOwner {
+    ChainSiloConfig memory remoteConfig = s_siloedChainConfigs[remoteChainSelector];
+
+    if (!remoteConfig.isSiloed) revert ChainNotSiloed(remoteChainSelector);
+
+    address oldRebalancer = remoteConfig.rebalancer;
+
+    s_siloedChainConfigs[remoteChainSelector].rebalancer = newRebalancer;
+
+    emit SiloedChainRebalancerSet(remoteChainSelector, newRebalancer, oldRebalancer);
+  }
+
+  /// @notice Sets the Rebalancer address for unsiloed chains.
+  /// @dev Only callable by the owner.
+  /// @param newRebalancer the address allowed to add liquidity for the given siloed chain.
```

Comment:
Probably good to add a `@dev` to say that 0 is a valid value and no 0 check is required?